applications/GeoMechanicsApplication/custom_strategies/schemes/newmark_quasistatic_U_Pw_scheme.hpp

```diff
@@ -116,14 +138,25 @@ class NewmarkQuasistaticUPwScheme
     {
         GeneralizedNewmarkScheme<TSparseSpace, TDenseSpace>::CheckAllocatedVariables(rModelPart);
 
-        for (const auto& r_node : rModelPart.Nodes()) {
-            this->CheckSolutionStepsData(r_node, DISPLACEMENT);
-            this->CheckSolutionStepsData(r_node, VELOCITY);
-            this->CheckSolutionStepsData(r_node, ACCELERATION);
-
-            this->CheckDof(r_node, DISPLACEMENT_X);
-            this->CheckDof(r_node, DISPLACEMENT_Y);
-            this->CheckDof(r_node, DISPLACEMENT_Z);
+        for (const auto& r_node : rModelPart.Nodes())
+        {
+            for (const auto& variable_derivative : mVariableDerivatives)
+            {
+                if (!rModelPart.HasNodalSolutionStepVariable(variable_derivative.instance))
+                    continue;
+
+                this->CheckSolutionStepsData(r_node, variable_derivative.instance);
+                this->CheckSolutionStepsData(r_node, variable_derivative.first_time_derivative);
+                this->CheckSolutionStepsData(r_node, variable_derivative.second_time_derivative);
+
+                std::vector<std::string> components{"X", "Y", "Z"};
```

Comment:
Thanks for pointing to this part, there is still a mistake in it: We should not check for "Z" here. since it should be allowed to be a 2D case. 

Since in that case these components with only XY are only used here and it's not really general, I don't think we gain a lot with storing it in every VariableWithTimeDerivatives.

Reply:
Agreed if it is only here then there is no need to make it general.